arrow-flight/tests/flight_sql_client_cli.rs

```diff
@@ -0,0 +1,552 @@
+        CommandGetSqlInfo, CommandGetTableTypes, CommandGetTables,
+        CommandPreparedStatementQuery, CommandPreparedStatementUpdate,
+        CommandStatementQuery, CommandStatementUpdate, ProstMessageExt, SqlInfo,
+        TicketStatementQuery,
+    },
+    utils::batches_to_flight_data,
+    Action, FlightData, FlightDescriptor, FlightEndpoint, FlightInfo, HandshakeRequest,
+    HandshakeResponse, IpcMessage, SchemaAsIpc, Ticket,
+};
+use arrow_ipc::writer::IpcWriteOptions;
+use arrow_schema::{ArrowError, DataType, Field, Schema};
+use assert_cmd::Command;
+use futures::Stream;
+use prost::Message;
+use tokio::{net::TcpListener, task::JoinHandle};
+use tonic::{Request, Response, Status, Streaming};
+
+const QUERY: &str = "SELECT * FROM table;";
+
+#[tokio::test(flavor = "multi_thread", worker_threads = 1)]
+async fn test_simple() {
+    let test_server = FlightSqlServiceImpl {};
+    let fixture = TestFixture::new(&test_server).await;
+    let addr = fixture.addr;
+
+    let stdout = tokio::task::spawn_blocking(move || {
+        Command::cargo_bin("flight_sql_client")
+            .unwrap()
+            .env_clear()
+            .env("RUST_BACKTRACE", "1")
+            .env("RUST_LOG", "warn")
+            .arg("--host")
+            .arg(addr.ip().to_string())
+            .arg("--port")
+            .arg(addr.port().to_string())
+            .arg(QUERY)
+            .assert()
+            .success()
+            .get_output()
+            .stdout
+            .clone()
+    })
+    .await
+    .unwrap();
+
+    fixture.shutdown_and_wait().await;
+
+    assert_eq!(
+        std::str::from_utf8(&stdout).unwrap().trim(),
+        "+--------------+-----------+\
+        \n| field_string | field_int |\
+        \n+--------------+-----------+\
+        \n| Hello        | 42        |\
+        \n| lovely       |           |\
+        \n| FlightSQL!   | 1337      |\
+        \n+--------------+-----------+",
+    );
+}
+
+const DEFAULT_TIMEOUT_SECONDS: u64 = 30;
+
+#[derive(Clone)]
+pub struct FlightSqlServiceImpl {}
+
+impl FlightSqlServiceImpl {
+    /// Return an [`FlightServiceServer`] that can be used with a
+    /// [`Server`](tonic::transport::Server)
+    pub fn service(&self) -> FlightServiceServer<Self> {
+        // wrap up tonic goop
+        FlightServiceServer::new(self.clone())
+    }
+
+    fn fake_result() -> Result<RecordBatch, ArrowError> {
+        let schema = Schema::new(vec![
+            Field::new("field_string", DataType::Utf8, false),
+            Field::new("field_int", DataType::Int64, true),
+        ]);
+
+        let mut string_builder = StringBuilder::new();
+        string_builder.append_value("Hello");
+        string_builder.append_value("lovely");
+        string_builder.append_value("FlightSQL!");
+
+        let mut int_builder = Int64Builder::new();
+        int_builder.append_value(42);
+        int_builder.append_null();
+        int_builder.append_value(1337);
```

Comment:
You could perhaps use the more modern 

(untested)
```suggestion
        let string_array = StringArray::from(["Hello", "lovely", "FlightSQL!");
        let int_array = Int64Array::from([Some(42), None, Some(1337)]);
```